core/src/main/java/tech/pegasys/eth2signer/core/multikey/DirectoryBackedArtifactSignerProvider.java

```diff
@@ -57,117 +54,73 @@ public DirectoryBackedArtifactSignerProvider(
+    LOG.info("Created artifactSigners = {}", artifactSigners.size());
   }
 
   @VisibleForTesting
-  protected LoadingCache<String, ArtifactSigner> getArtifactSignerCache() {
-    return artifactSignerCache;
-  }
-
-  private Optional<ArtifactSigner> loadSignerForIdentifier(final String signerIdentifier) {
-    final Filter<Path> pathFilter = signerIdentifierFilenameFilter(signerIdentifier);
-    final Collection<ArtifactSigner> matchingSigners = findSigners(pathFilter, signerParser::parse);
-    if (matchingSigners.size() > 1) {
-      LOG.error(
-          "Found multiple signing metadata file matches for signer identifier " + signerIdentifier);
-      return Optional.empty();
-    } else if (matchingSigners.isEmpty()) {
-      return Optional.empty();
-    } else {
-      return Optional.of(matchingSigners.iterator().next());
-    }
+  protected Map<String, ArtifactSigner> getArtifactSignerCache() {
+    return artifactSigners;
   }
 
   private <T> Collection<T> findSigners(
-      final DirectoryStream.Filter<? super Path> filter, final Function<Path, T> mapper) {
-    final Collection<T> signers = new HashSet<>();
-
-    try (final DirectoryStream<Path> directoryStream =
-        Files.newDirectoryStream(configsDirectory, filter)) {
-      for (final Path file : directoryStream) {
-        try {
-          signers.add(mapper.apply(file));
-        } catch (final Exception e) {
-          renderException(e, file.getFileName().toString());
-        }
-      }
-      return signers;
-    } catch (final IOException | SecurityException e) {
-      LOG.warn("Error searching for signing metadata files: {}", e.getMessage());
-      return Collections.emptySet();
+      final Predicate<? super Path> filter, final Function<Path, T> mapper) {
+    try (final Stream<Path> fileStream = Files.list(configsDirectory)) {
+      LOG.info("Config files count = {}", fileStream.count());
+    } catch (final IOException e) {
+      LOG.error("Didn't want this one", e);
     }
-  }
 
-  private Filter<Path> signerIdentifierFilenameFilter(final String signerIdentifier) {
-    return entry -> {
-      final String baseName = FilenameUtils.getBaseName(entry.toString());
-      return matchesFileExtension(entry)
-          && baseName.toLowerCase().endsWith(signerIdentifier.toLowerCase());
-    };
+    try (final Stream<Path> fileStream = Files.list(configsDirectory)) {
+      return fileStream
+          .parallel()
+          .filter(filter)
+          .map(
+              signerConfigFile -> {
+                try {
+                  return mapper.apply(signerConfigFile);
+                } catch (final Exception e) {
+                  renderException(e, signerConfigFile.getFileName().toString());
+                  return null;
+                }
+              })
+          .filter(Objects::nonNull)
+          .collect(Collectors.toSet());
+    } catch (final IOException e) {
+      LOG.error("Didn't want this one", e);
```

Comment:
Can improve this error message a bit I think. Perhaps something like "Error searching for signing metadata files"?

Reply:
agreed - updated